File: Java/0034_find_first_and_last_position_of_element_in_sorted_array/0034_find_first_and_last_position_of_element_in_sorted_array.java
class Solution {
    public int[] searchRange(int[] nums, int target) {
        int left = 0;
        int right = nums.length - 1;
        while (left <= right) {
            int mid = left + (right - left) / 2;
            if (nums[mid] == target) {
                left = right = mid;
                while (left >= 0 && nums[left] == nums[mid])
                    left -= 1;
                while (right <= nums.length - 1 && nums[right] == nums[mid])
                    right += 1;
                int[] found = {left + 1, right - 1};
                return found;
            }
            else if (nums[mid] < target)
                left = mid + 1;
            else if (nums[mid] > target)
                right = mid - 1;
        }
        int[] notFound = {-1, -1};
        return notFound; 
    }
}

File: Python/0034_find_first_and_last_position_of_element_in_sorted_array/0034_find_first_and_last_position_of_element_in_sorted_array.py
class Solution:
    def searchRange(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        left = 0
        right = len(nums) - 1
        while left <= right:
            mid = left + (right - left) // 2
            if nums[mid] == target:
                left = right = mid
                while left >= 0 and nums[left] == nums[mid]:
                    left -= 1
                while right <= len(nums) - 1 and nums[right] == nums[mid]:
                    right += 1
                return [left + 1, right - 1]
            elif nums[mid] < target:
                left = mid + 1
            elif nums[mid] > target: 
                right = mid - 1
        return [-1, -1]
